src/views/acls/UserLoginACL/UserLoginACLCreateUpdate.vue

```diff
@@ -81,6 +81,13 @@ export default {
         return url
       },
       cleanFormValue(value) {
+        if (
+          Array.isArray(value.rules.time_period) &&
+          value.rules.time_period.every(item => item.value === '')
+        ) {
+          value.rules.time_period = []
+        }
+
         if (!Array.isArray(value.rules.ip_group)) {
           value.rules.ip_group = value.rules.ip_group ? value.rules.ip_group.split(',') : []
         }
```

Comment:
There are no major discrepancies between the source of the code snippet in 2021 and the version you're providing today. The comments in the lines where the changes were made suggest that this is meant to be optimized for performance rather than maintainability, but I'll assume it isn't needed for discussion purposes.

However, here's an optimization suggestion in terms of readability:

```
if (value.rules.time_period && value.rules.time_period.length > 0 && !value.rules.time_period.every(item => item && item.value === '')) {
  // Remove empty items from time period rules
// ...
}
```